engine/src/main/java/org/pentaho/di/trans/steps/databaselookup/DatabaseLookup.java

```diff
@@ -679,4 +692,183 @@ private void connectDatabase( Database database ) throws KettleDatabaseException
+      response.put("actionStatus", "FAILURE");
+      response.put("error", "Method not found: " + fieldName);
+    } catch (InvocationTargetException e) {
+      log.logError("Error invoking method: " + e.getCause().getMessage(), e.getCause());
+      response.put("actionStatus", "FAILURE");
+      response.put("error", "Error invoking method: " + fieldName);
+    } catch (IllegalAccessException e) {
+      log.logError("Illegal access to method: " + e.getMessage(), e);
+      response.put("actionStatus", "FAILURE");
+      response.put("error", "Illegal access to method: " + fieldName);
+    } catch (Exception e) {
+      log.logError("Unexpected error: " + e.getMessage(), e);
+      response.put("actionStatus", "FAILURE");
+      response.put("error", "Unexpected error occurred");
+    }
+    return response;
+  }
+
+  @SuppressWarnings("java:S1144") // Using reflection this method is being invoked
+  private JSONObject getSchemaAction(Map<String, String> queryParams) {
+    JSONObject response = new JSONObject();
+    response.put("actionStatus", "FAILURE");
+
+    String connectionName = queryParams.get("connection");
+    if (connectionName == null || connectionName.isBlank()) {
+      response.put("error", "Connection name is missing or empty.");
+      return response;
+    }
+
+    try {
+      String[] schemas = getSchemaNames(connectionName);
+      JSONArray schemaNames = Arrays.stream(schemas)
+        .collect(JSONArray::new, JSONArray::add, JSONArray::addAll);
+
+      response.put("schemaNames", schemaNames);
+      response.put("actionStatus", "SUCCESS");
+    } catch (KettleDatabaseException e) {
+      log.logError("Error fetching schema names: " + e.getMessage(), e);
+      response.put("error", "Database error occurred: " + e.getMessage());
+    } catch (Exception e) {
+      log.logError("Unexpected error: " + e.getMessage(), e);
+      response.put("error", "An unexpected error occurred.");
+    }
+    return response;
+  }
+
+  private String[] getSchemaNames(String dbname) throws KettleDatabaseException {
+    DatabaseMeta databaseMeta = Optional.ofNullable(getTransMeta().findDatabase(dbname))
+      .orElseThrow(() -> new KettleDatabaseException("Database not found: " + dbname));
+
+    LoggingObjectInterface loggingObject = new SimpleLoggingObject(
+      "DB Lookup Step", LoggingObjectType.STEP, null);
+
+    try (Database database = new Database(loggingObject, databaseMeta)) {
+      database.connect();
+      return Optional.ofNullable(database.getSchemas())
+        .map(Const::sortStrings)
+        .orElse(new String[0]);
+    } catch (Exception e) {
+      log.logError("Error fetching schemas for database: " + dbname, e);
+      throw new KettleDatabaseException("Failed to fetch schemas for database: " + dbname, e);
+    }
+  }
+
+  @SuppressWarnings("java:S1144") // Using reflection this method is being invoked
+  private JSONObject getTableFieldAction(Map<String, String> queryParams) {
```

Comment:
Does the same code is present in dialog as well? If so then we have to refactor the dialog code for maintainability